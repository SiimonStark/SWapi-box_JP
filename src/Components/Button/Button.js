import React from 'react';
import PropTypes from 'prop-types';

const Button = ({name, updateTab}) => {
  return (
    <button id={name} onClick={updateTab}>
      {name}
    </button>
  )
}

Button.propTypes = {
  name: PropTypes.string.isRequired,
  updateTab: PropTypes.func.isRequired
}

export default Button;