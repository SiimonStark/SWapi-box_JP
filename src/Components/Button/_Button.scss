nav button {
  margin: 0 35px;
  color: $OGYellow;
  background: rgba(black, .25);
  border: $OGYellow 3px solid;
  border-radius: 20px;

  &:hover {
    background-color: $OGYellow;
    color: black;
  }
}

.button--active {
  background-color: $OGYellow;
  color: black;
}