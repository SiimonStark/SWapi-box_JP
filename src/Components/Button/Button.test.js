import React from 'react';
import {shallow} from 'enzyme';
import Button from './Button';

let updateTabMock = jest.fn();

describe('Button', ()=>{
  let wrapper;
  beforeEach(()=>{
    wrapper = shallow(
      <Button name={'People'} updateTab={updateTabMock}/>
    );
  });

  it('should match snapshot', ()=>{
    expect(wrapper).toMatchSnapshot();
  });

  it('onClick should call updateTab', ()=>{
    let button = wrapper.find('button');
    button.simulate('click');
    expect(updateTabMock).toHaveBeenCalled();
  });
});