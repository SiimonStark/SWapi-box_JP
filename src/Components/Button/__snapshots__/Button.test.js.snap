// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<button
  id="People"
  onClick={[MockFunction]}
>
  People
</button>
`;
